scripts/process_uploads.py

```diff
@@ -0,0 +1,161 @@
+#!/usr/bin/env python3
+"""
+Processes a new batch of samples from the UPLOAD bucket,
+into the MAIN and ARCHIVE buckets using the batch_move_files function. 
+
+Determines the files to be moved through a comparison of the 
+most recently updated CSV and the CSV uploaded in a previous batch run. 
+"""
+
+import os
+import io
+import csv
+from os.path import join, basename
+from typing import List
+import click
+import hail as hl
+import hailtop.batch as hb
+from google.cloud import storage
+from joint_calling.upload_processor import batch_move_files
+
+
+def samples_from_csv(bucket_name, prefix):
+    """Determines list of samples from a csv file within a specified bucket
+
+    Assumptions
+    ==========
+    Only one .csv file exists in the upload bucket and previous batch directory.
+    """
+
+    client = storage.Client()
+    bucket = client.get_bucket(bucket_name)
+
+    all_blobs = list(client.list_blobs(bucket_name, prefix=prefix))
+    csv_path = list(filter(lambda blob: blob.name.endswith('.csv'), all_blobs))[0].name
```

Comment:
There's a little Python trick to not produce the full list, if you're only interested in the first item:

```suggestion
    csv_path = next(filter(lambda blob: blob.name.endswith('.csv'), all_blobs)).name
```